src/Common/src/CoreLib/System/HashCode.cs

```diff
@@ -65,12 +69,38 @@ public struct HashCode
         private uint _queue1, _queue2, _queue3;
         private uint _length;
 
+#if BUILDING_COMPAT_HASHCODE // Will only be defined when building Microsoft.Bcl.HashCode project
+        private static unsafe uint GenerateGlobalSeed()
+        {
+            if (UseNonRandomizedGlobalSeed)
+            {
+                uint result;
+                Interop.GetRandomBytes((byte*)&result, sizeof(uint));
```

Comment:
Since the problem is that the Interop.GetRandomBytes defined for the package doesn't actually do any work, I'd move the fix to in that method, and then this file can avoid more (any?) \#ifs.

```C#
    internal static unsafe void GetRandomBytes(byte* buffer, int length)
    {
        if (!LocalAppContextSwitches.UseNonRandomizedHashSeed)
        {
            byte[] tmp = new byte[length];

            using (RandomNumberGenerator rng = RandomNumberGenerator.Create())
            {
                rng.GetBytes(tmp);
                Marshal.Copy(tmp, 0, (IntPtr)buffer, length);
            }
        }
    }